### PR TITLE
Add recipe for jigsawpy

### PR DESCRIPTION
**Pull request by xylar**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/jigsawpy/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    simulation. JIGSAW includes both Delaunay-refinement based algorithms for
+    the construction of new meshes, as well as optimisation driven methods for
+    the improvement of existing grids.
+  doc_url: https://github.com/dengwirda/jigsaw-python
+  dev_url: https://github.com/dengwirda/jigsaw-python
+
+extra:
+  recipe-maintainers:
+    - xylar
+    - dengwirda
```
Comment (xylar) [inline recipes/jigsawpy/meta.yaml]: @dengwirda, can you confirm that you are willing to be a maintainer of this recipe?

Comment (dengwirda) [reply recipes/jigsawpy/meta.yaml]: Yes, of course @xylar!

**Comment by xylar** [comment]
@dengwirda, under Windows, we're getting:
```
Traceback (most recent call last):
  File "example.py", line 6, in <module>
    from tests.case_0_ import case_0_
  File "C:\Miniconda\conda-bld\jigsawpy_1581616685544\test_tmp\tests\case_0_.py", line 13, in <module>
    import jigsawpy
  File "C:\Miniconda\conda-bld\jigsawpy_1581616685544\_test_env\lib\site-packages\jigsawpy\__init__.py", line 58, in <module>
    from jigsawpy import jigsaw, libsaw
  File "C:\Miniconda\conda-bld\jigsawpy_1581616685544\_test_env\lib\site-packages\jigsawpy\libsaw.py", line 74, in <module>
    JLIB = ct.cdll.LoadLibrary(str(JLIBNAME))
  File "C:\Miniconda\conda-bld\jigsawpy_1581616685544\_test_env\lib\ctypes\__init__.py", line 451, in LoadLibrary
    return self._dlltype(name)
  File "C:\Miniconda\conda-bld\jigsawpy_1581616685544\_test_env\lib\ctypes\__init__.py", line 373, in __init__
    self._handle = _dlopen(self._name, mode)
FileNotFoundError: Could not find module 'jigsaw.dll'. Try using the full path with constructor syntax
```
I'm not very good at debugging windows and don't have a windows machine for local testing, but I'll see what I can figure out.  But wondered if you have any ideas.  It could be that the DLL isn't in the search path or that the name isn't right.

**Comment by xylar** [comment]
What I've figured out so far is that https://github.com/conda-forge/jigsaw-feedstock/ installs `jigsaw.dll` into a `lib` folder in the conda environment, whereas other packages (e.g. `libnetcdf`) install DLLs into `Library/lib`.  My guess is we'll need to do a new build of `jigsaw` that fixes this before we can get this recipe working.

**Comment by xylar** [comment]
@conda-forge-admin, please restart ci

**Comment by xylar** [comment]
Nope, I got it wrong.  Under Windows, `.lib` files end up in `Library/lib` but `.dll` files end up in `Library/bin`.  I'll see how other libraries handle this.

**Comment by xylar** [comment]
@conda-forge-admin, please restart ci

**Comment by xylar** [comment]
@dengwirda, nothing I have tried has fixed the windows build :-(

Could you try creating a conda environment with `jigsaw` on your windows machine and then running the `jigsawpy` tests with it to see if you see the same error?  Creating and activating the conda environment is just:
```
conda create -n test jigsaw=0.9.12 python=3.7 numpy pathlib
conda activate test
```

**Comment by dengwirda** [comment]
@xylar okay, I've done some spelunking on `win`, and I think this may relate to setting the Windows  `path` system variable.

In a workflow where:
- I already have `jigsaw` installed (the `c++` binary part) in some directory `DIR`.
- and then install only the `python` part of `jigsawpy` (i.e. just running `python setup.py install`).

I have to add `DIR` to the Windows `path` variable to enable `jigsawpy` to find the `jigsaw` binaries. 

I'm still trying to figure out exactly how all of this works via `conda`, but if `jigsaw-conda` copies all of the binaries to `Library/bin`, hopefully appending `Library/bin` to the Windows `path` might do the trick...

**Comment by xylar** [comment]
Thanks @dengwirda.  I'll look into that.  I kind of wonder how Windows finds the DLLs for other libraries in the same `Library/bin` directory.  We can set `path` with an `activate.bat` but most libraries don't seem to need this.  Maybe it's specific to the python interfacing.

**Comment by dengwirda** [comment]
@xylar okay, I've got `miniconda` up and running on `win` now, and it seems that there's more to this story.

It seems that `miniconda` installs an "anaconda cmd prompt" for use in Windows, and that when you run from this cmd-line (rather than the standard Windows one) it seems to add various directories (including `Library/bin` etc) to the Windows path auto-magically.

If I install the current `jigsawpy` (python setup.py install) and load
````
conda create -n test jigsaw=0.9.12 python=3.7 numpy pathlib
conda activate test
````
this actually just works for me already.

The business about the "anaconda cmd-line" and the `path` variable is still a little confusing to me though. Is it possible to dump out 
````
echo %PATH%
````
during the tests on `win` and see if the `path` variable includes the `Library/bin` directory we need?

**Comment by xylar** [comment]
I can reproduce the error on my Windows virtual machine so I'm going to play around there.  I'll see what the `path` variable contains and also see if explicitly passing an absolute path to the library helps.

**Comment by xylar** [comment]
@conda-forge/python, we would appreciate a review of this recipe at your convenience.

Appveyor wasn't working earlier and seems to be stuck queuing for now.  Perhaps that's fine since this is a `noarch: python` recipe?

**Comment by scopatz** [comment]
Yeah, noarch only needs to run on linux

File: recipes/jigsawpy/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python >=3.6
+    - pip
+    - setuptools
```
Comment (scopatz) [inline recipes/jigsawpy/meta.yaml]: I don't think you need both pip and setuptools.  Pip is all that is needed

File: recipes/jigsawpy/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  run:
+    - python >=3.6
+    - jigsaw >=0.9.12
+    - numpy
+    - pathlib
+
+test:
+  requires:
+    - matplotlib
```
Comment (scopatz) [inline recipes/jigsawpy/meta.yaml]: Can this be `matplotlib-base`

Comment (xylar) [reply recipes/jigsawpy/meta.yaml]: Ah, good point. I believe it can be matplotlib-base. I'll change this.

File: recipes/jigsawpy/meta.yaml (hunk 2 above)
Comment (xylar) [reply recipes/jigsawpy/meta.yaml]: Okay, easy enough to remove, thanks!

File: recipes/jigsawpy/meta.yaml (hunk 3 above)
Comment (xylar) [reply recipes/jigsawpy/meta.yaml]: done!

File: recipes/jigsawpy/meta.yaml (hunk 2 above)
Comment (xylar) [reply recipes/jigsawpy/meta.yaml]: done!

**Comment by xylar** [comment]
Side effect: AppVeyor is working again.

**Comment by scopatz** [comment]
Thanks!

**Comment by xylar** [comment]
Thanks again!

**Pull request merged**